solidity/ecdsa/contracts/libraries/EcdsaAuthorization.sol

```diff
@@ -15,25 +15,512 @@
+
+        if (operator == address(0)) {
+            // Operator is not known. It means `registerOperator` was not
+            // called yet, and there is no chance the operator could
+            // call `joinSortitionPool`. We can just ignore this update because
+            // operator was never in the sortition pool.
+            return;
+        } else {
+            // Operator is known. It means that this operator is or was in the
+            // sortition pool and the sortition pool may need to be updated.
+            //
+            // If the sortition pool is not locked and the operator is in the
+            // sortition pool, we are updating it.
+            //
+            // To keep stakes synchronized between applications when staking
+            // providers are slashed, without the risk of running out of gas,
+            // the staking contract queues up slashings and let users process
+            // the transactions. When an application slashes one or more staking
+            // providers, it adds them to the slashing queue on the staking
+            // contract. A queue entry contains the staking provider’s address
+            // and the amount they are due to be slashed.
+            //
+            // When there is at least one staking provider in the slashing
+            // queue, any account can submit a transaction processing one or
+            // more staking providers' slashings, and collecting a reward for
+            // doing so. A queued slashing is processed by updating the staking
+            // provider’s stake to the post-slashing amount, updating authorized
+            // amount for each affected application, and notifying all affected
+            // applications that the staking provider’s authorized stake has
+            // been reduced due to slashing.
+            // 
+            // The entire idea is that the process transaction is expensive
+            // because each application needs to be updated, so the reward for
+            // the processor is hefty and comes from the slashed tokens.
+            // Practically, it means that if the sortition pool is unlocked, and
+            // can be updated, it should be updated because we already paid
+            // someone for updating it.
+            //
+            // If the sortition pool is locked, update needs to wait. Other
+            // sortition pool members are incentivized to call
+            // `updateOperatorStatus` for the problematic operator because they
+            // will increase their rewards this way.
+            if (
+                sortitionPool.isOperatorInPool(operator) &&
+                !sortitionPool.isLocked()
+            ) {
+                updateOperatorStatus(
+                    self,
+                    tokenStaking,
+                    sortitionPool,
+                    operator
+                );
+            }
+        }
+
+        // TODO: event?
```

Comment:
@nkuba I think we should emit an event but only if the pool could not be updated because it was locked. What do you think? 

The idea behind events in `WalletRegistry` is not to duplicate those from `TokenStaking` but to make client implementation easier. And the client should act only if the pool could not be updated.

Reply:
I'm on a fence here. It would be easier and more consistent to emit an event any time the decrease happens.

After receiving an event in the client we will have to verify if we need to submit a `updateOperatorStatus` transaction or not anyways.

(side note: I'm curious how will we solve the problem of all the clients calling this function at once)

Reply:
As for the event emission - yes, we should do it. I'm just not sure which one of the two possibilities will be better. So leaving it up to you. Both are fine.